docs/dev/testing/sdk-integration-tests.md

```diff
@@ -1,9 +1,511 @@
 >
-> This document will be populated in Phase 4 of milestone ls-test-improvement (#556).
->
-> See issue #560 for progress.
+> This document is part of the centralized testing documentation suite. See `@docs/dev/testing/README.md` for the complete TOC.
+
+## Overview
+
+The SDK contains integration tests that make real API calls to LangSmith and other LangChain services. These tests verify end-to-end functionality of the SDK against live APIs.
+
+## Key Concepts
+
+### Deployment vs Revision Status
+
+LangGraph Cloud has two distinct status types - see [langgraph-deployments-and-revisions.md](../../langgraph-deployments-and-revisions.md) for details:
+
+- **DeploymentStatus** (e.g., `Ready`) - Overall deployment state
+- **RevisionStatus** (e.g., `Deployed`) - Build/deploy state of a specific revision
+
+Test fixtures wait for `RevisionStatus::Deployed`, not `DeploymentStatus::Ready`.
+
+### Test Deployment Naming
+
+Integration tests use exactly **two deployment types** with standardized naming:
+
+| Type | Pattern | Behavior |
+|------|---------|----------|
+| **PR/Dev** | `pr-integration-test-{timestamp}` | Get-or-create by prefix, cleaned by cron (4hr threshold) |
+| **Release** | `release-integration-test-{timestamp}` | Always fresh, self-deleting |
+
+**Using TestDeploymentConfig:**
+- `TestDeploymentConfig::default()` - Creates PR/dev config with prefix-based reuse
+- `TestDeploymentConfig::for_release_tests()` - Creates release config (always fresh)
+
+The `get_or_create_deployment()` function handles:
+- **With prefix**: Searches for existing `pr-integration-test-*` deployment, reuses if found
+- **Without prefix**: Always creates a new deployment (for release tests)
+
+## Running Integration Tests
+
+Integration tests are marked with `#[ignore]` and require API keys to run.
+
+### Prerequisites
+
+Set the `LANGSMITH_API_KEY` environment variable:
+
+```bash
+export LANGSMITH_API_KEY="<your-api-key>"
+```
+
+### Run All Integration Tests
+
+```bash
+# Run all ignored (integration) tests
+cargo test --test integration_test -- --ignored --nocapture
+
+# Run specific integration test
+cargo test --test integration_test test_list_prompts_from_prompthub -- --ignored --nocapture
+```
+
+### Run Unit Tests Only (Default)
+
+```bash
+# This skips integration tests (they're ignored by default)
+cargo test
+```
+
+### Run Everything
+
+```bash
+# Run both unit and integration tests
+cargo test -- --include-ignored
+```
+
+## Available Integration Tests
+
+### PromptHub Tests
+
+#### `test_list_prompts_from_prompthub` ✅
+
+**Status**: Working
+
+Tests the ability to list prompts from the LangSmith PromptHub.
+
+**Requirements**:
+- Valid `LANGSMITH_API_KEY`
+- Read permissions
+
+**What it tests**:
+- Authentication with LangSmith API
+- Fetching and parsing paginated prompt list
+- API response deserialization
+
+#### `test_push_prompt_to_prompthub` ✅
+
+**Status**: Working (requires prompt repository to exist first)
```

Comment:
this is misleading. This test should test pushing a private prompt to the prompthub, which uses the null repository `-`.